File: BackEnd/backend/src/main/java/com/example/backend/entities/contest/ContestParticipation.java
package com.example.backend.entities.contest;

import com.example.backend.entities.content.MediaFile;
import com.example.backend.entities.users.User;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "contest_participation")
public class ContestParticipation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "participant_id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "contest_id", referencedColumnName = "contest_id")
    private Contest contest;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToOne
    @JoinColumn(name = "file_id")
    private MediaFile mediaFile;

    @Embedded
    private QuoteCriterion quoteCriterion;


}

File: Frontend/src/app/core/interfaces/contest.participation.ts
import { Contest } from "./contest"
import { MediaFile } from "./media.file"
import { User } from "./user"

export interface ContestParticipation {
    id: number
    participant: User,
    contest: Contest,
    mediaFile?: MediaFile, 
    quoteCriterion: QuoteCritirion
}

export interface QuoteCritirion {
    vote: number,
    description: string,
    isQuoted: boolean
}
